lib/LaTeXML/Engine/TeX_Box.pool.ltxml

```diff
@@ -331,14 +362,32 @@ Tag('svg:foreignObject', autoOpen => 1, autoClose => 1,
             return; } } } }
     # Otherwise, we've still got an svg:foreignObject;
     # Make sure we get a size, in case autoOpen'd
-    if ($whatsit) {
+    # Reminder: TeX/LaTeXML think origin is left & baseline
+    # svg foreignObject content will draw down from the top
+    if ($whatsit && !$node->getAttribute('_wrapped')) {
       my ($w, $h, $d) = $whatsit->getSize;
-      my $y  = $STATE->lookupDefinition(T_CS('\baselineskip'))->valueOf->pxValue;
-      my $ht = $h->add($d);
-      $node->setAttribute(width     => $w->pxValue)  unless $node->hasAttribute('width');
-      $node->setAttribute(height    => $ht->pxValue) unless $node->hasAttribute('height');
-      $node->setAttribute(transform => "matrix(1 0 0 -1 0 $y)");
-      $node->setAttribute(overflow  => 'visible'); } });
+      my $H = $h->add($d);
+      my $W = $w;
+      my $x = Dimension(0);
+      my $y = $h;
+      Debug("SVG:foreignObject ".$w->pxValue.' x '.$h->pxValue.' x '.$d->pxValue."; "
+        . " => H=".$H->pxValue." @ y=".$y->pxValue
+        . " for Whatsit=".ToString($whatsit)) if $LaTeXML::DEBUG{svg_verbose};
+      # Size of FO is width and total height (+ depth)
+      $node->setAttribute(width     => $W->pxValue) unless $node->hasAttribute('width');
+      $node->setAttribute(height    => $H->pxValue) unless $node->hasAttribute('height');
+      # Set top of FO height above current, as if baseline will line up w/outside
+      $node->setAttribute(transform => 'matrix(1 0 0 -1 '.$x->pxValue.' '.$y->pxValue.')');
+      $node->setAttribute(overflow  => 'visible');
+      $node->setAttribute('_wrapped'=>1);
+      # Add a wrapper, so that CSS has a chance of tweaking/centering
+      my $wrapper = $document->wrapNodes('ltx:text',$node->childNodes);
+      $document->addClass($wrapper, 'ltx_foreignObject_wrapper');
```

Comment:
Interesting that you are adding this to the XML!

Reply:
tiny naming nit: @brucemiller could we avoid that capital `O`? 

It matches the SVG element name, but makes our CSS class suite noisier. either just `ltx_foreignobject_wrapper` or `ltx_foreign_object_wrapper` would be fine.

Reply:
Yeah, I was surprised that I left it that way, too! Thanks for calling me out. I moved the wrapping to XSLT, which is more appropriate, and you originally had it. And I went ahead & made 2 wrappers, with your class names, as you had it.